src/webhook/pkg/mutating_webhook/remove-resourcespec.go

```diff
@@ -26,18 +35,98 @@ func (a *PodMutator) mutatePod(pod *corev1.Pod) (*corev1.Pod, error) {
 		return mpod, nil
 	}
 
+	mpod = adjustResourceSpec(mpod)
+
+	return mpod, nil
+}
+
+// function to remove resource spec from the pod spec
+// add the cumulative resources as annotation to pod spec
+// add the peer-pod resource to the first container in the pod spec
+
+func adjustResourceSpec(pod *corev1.Pod) *corev1.Pod {
+
+	// Get total CPU resource requests
+	cpuRequest := utils.GetResourceRequestQuantity(pod, corev1.ResourceCPU)
+
+	// Get total CPU resource limits
+	cpuLimit := utils.GetResourceLimitQuantity(pod, corev1.ResourceCPU)
+
+	// Get total Memory resource requests
+	memoryRequest := utils.GetResourceRequestQuantity(pod, corev1.ResourceMemory)
+
+	// Get total Memory resource limits
+	memoryLimit := utils.GetResourceLimitQuantity(pod, corev1.ResourceMemory)
+
+	// Get total GPU resource requests
+	gpuRequest := utils.GetResourceRequestQuantity(pod, corev1.ResourceName(GPU_RESOURCE_NAME))
+
+	// log the resource values
+	logger.Printf("CPU Request: %s, CPU Limit: %s, Memory Request: %s, Memory Limit: %s, GPU Request: %s",
+		cpuRequest.String(), cpuLimit.String(), memoryRequest.String(), memoryLimit.String(), gpuRequest.String())
+
+	logger.Printf("Memory: %s", memoryRequest.String())
+	// GPU resources are always requested in whole numbers and request and limits are same.
+	// So we don't need to check for limits
+
+	// Add the cumulative resources as annotation to pod spec
+	// Use cpuLimit and memoryLimit if those are greater than cpuResource and memoryResource
+	annotations := pod.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	// A non-existent request or limit will be 0 and we don't need to add annotation for 0
+	// We only add annotation if the value is greater than 0
+	// Limit will always be preferred over request
+
+	// Add cpu annotation
+	if cpuRequest.Cmp(resource.MustParse("0")) != 0 && cpuLimit.Cmp(cpuRequest) >= 0 {
```

Comment:
Could these be made more readable with https://github.com/kubernetes/apimachinery/blob/v0.30.2/pkg/api/resource/quantity.go#L520?
e.g. 
```
if !cpuRequest.isZero() && cpuLimit.Cmp(cpuRequest) >= 0 {
```
...
Assume that has the same effect as the current code?
I appreciate it's not much, but it took my tired brain a while to process `Cmp(resource.MustParse("0")) != 0`!

Reply:
Good suggestion :-) ..